lib/dice/src/handoff.rs

```diff
@@ -0,0 +1,90 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use crate::{AliasCert, AliasOkm, DeviceIdSelfCert};
+use hubpack::SerializedSize;
+use lpc55_pac::syscon::RegisterBlock;
+use serde::{Deserialize, Serialize};
+
+// This memory is the USB peripheral SRAM that's 0x4000 bytes long. Changes
+// to this address must be coordinated with the [dice_*] tables in
+// chips/lpc55/chip.toml
+// TODO: get from app.toml -> chip.toml at build time
+const MEM_START: usize = 0x4010_0000;
+const ALIAS_START: usize = MEM_START;
+
+/// The Handoff type is a thin wrapper over the memory region used to transfer
+/// DICE artifacts (seeds & certs) from stage0 to hubris tasks. It is intended
+/// for use by stage0 to write these artifacts to memory where they will later
+/// be read out by a hubris task.
+pub struct Handoff<'a>(&'a RegisterBlock);
+
+impl<'a> Handoff<'a> {
+    pub fn turn_on(syscon: &'a RegisterBlock) -> Self {
+        // handoff through USB SRAM requires we power it on
+        syscon.ahbclkctrl2.modify(|_, w| w.usb1_ram().enable());
+        syscon
+            .presetctrl2
+            .modify(|_, w| w.usb1_ram_rst().released());
+
+        Self(syscon)
+    }
+
+    pub fn turn_off(self) {
+        self.0
+            .presetctrl2
+            .modify(|_, w| w.usb1_ram_rst().asserted());
+        self.0.ahbclkctrl2.modify(|_, w| w.usb1_ram().disable());
+    }
+
+    pub fn store(&self, alias_data: &AliasData) -> usize {
+        let dst_ptr = ALIAS_START as *mut [u8; AliasData::MAX_SIZE];
+        // SAFETY: Dereferencing this raw pointer is necessary to write to the
+        // memory region used to handoff DICE artifacts to Hubris tasks. This
+        // pointer will references a valid memory region provided two
+        // conditions are met:
+        // 1) The associated memory region has been enabled / turned on if
+        // necessary. This happens in the constructor / 'turn_on' function.
+        // 2) The function call is made by code in stage0.
+        let dst: &mut [u8] = unsafe { &mut *dst_ptr };
```

Comment:
```suggestion
        let dst: &mut [u8] = unsafe { core::slice::from_raw_parts_mut(ALIAS_START as *mut u8, AliasData::MAX_SIZE) };
```

Reply:
Should also check that AliasData::MAX_SIZE fits in the dice_alias peripheral's range.  That should be a compile-time check of some sort.

Reply:
https://github.com/oxidecomputer/hubris/issues/751